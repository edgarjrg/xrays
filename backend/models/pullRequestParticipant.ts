import { Model, Column, Table, UpdatedAt, CreatedAt, DataType } from "sequelize-typescript";

@Table({ tableName: 'PULL_REQUEST_PARTICIPANT' })
export class PullRequestParticipant extends Model<PullRequestParticipant> {

  @Column({ field: 'date' })
  date?: string;

  @Column({ field: 'pull_request_url' })
  pullRequestURL?: string;

  @Column({ field: 'author' })
  author?: string;

  @Column({ field: 'type' })
  type?: string;

  @CreatedAt
  @Column({ field: 'created_at', type: DataType.DATE })
  createdAt?: Date;

  @UpdatedAt
  @Column({ field: 'updated_at', type: DataType.DATE })
  updatedAt?: Date;
}

export default PullRequestParticipant