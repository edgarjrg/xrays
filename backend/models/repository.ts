import { Model, Column, Table, UpdatedAt, CreatedAt, DataType } from "sequelize-typescript";

@Table({ tableName: 'REPOSITORY' })
export class Repository extends Model<Repository> {

  @Column({ field: 'project_key' })
  projectKey?: string;

  @Column({ field: 'repo_slug' })
  repoSlug?: string;

  @Column({ field: 'user_name' })
  userName?: string;

  @CreatedAt
  @Column({ field: 'created_at', type: DataType.DATE })
  createdAt?: Date;

  @UpdatedAt
  @Column({ field: 'updated_at', type: DataType.DATE })
  updatedAt?: Date;
}

export default Repository