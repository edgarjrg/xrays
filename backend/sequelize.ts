import { Sequelize } from 'sequelize-typescript';

export function db() {
  return new Sequelize({
    dialect: 'postgres',
    database: 'xrays',
    username: 'xrays',
    password: '123456',
    modelPaths: [__dirname + '/models']
  });
}