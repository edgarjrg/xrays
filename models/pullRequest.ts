import { Model, Column, Table, UpdatedAt, CreatedAt, DataType } from "sequelize-typescript";

@Table({ tableName: 'PULL_REQUEST' })
export class PullRequest extends Model<PullRequest> {

  @Column({ field: 'project_key' })
  projectKey?: string;

  @Column({ field: 'pull_request_url' })
  pullRequestURL?: string;

  @Column({ field: 'user_name' })
  userName?: string;

  @Column({ field: 'repo_slug' })
  repoSlug?: string;

  @CreatedAt
  @Column({ field: 'created_at', type: DataType.DATE })
  createdAt?: Date;

  @UpdatedAt
  @Column({ field: 'updated_at', type: DataType.DATE })
  updatedAt?: Date;
}

export default PullRequest