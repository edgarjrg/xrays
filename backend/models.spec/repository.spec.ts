'use strict'

import { db } from '../sequelize'
import { Repository } from "../models/repository";
import { omit, map } from 'ramda';

const repository = {
  projectKey: "atlassian-oauth",
  repoSlug: "atlassian-oauth",
  userName: "atlassian",
}

describe('repository @database', () => {
  beforeAll(() => {
    const sequelize = db()
    sequelize.models.Repository.destroy({ truncate: true, cascade: true })
  })

  it('should read an write', () => {

    return Repository
      .create(repository)
      .then(response => response.dataValues)
      .then(omit(['id', 'createdAt', 'updatedAt']))
      .then(result => {
        expect(result).toMatchSnapshot()
      })
  })
})
