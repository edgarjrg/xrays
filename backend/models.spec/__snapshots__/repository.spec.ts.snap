// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`repository @database should read an write 1`] = `
Object {
  "projectKey": "atlassian-oauth",
  "repoSlug": "atlassian-oauth",
  "userName": "atlassian",
}
`;
